core/domain/stats_services_test.py

```diff
@@ -1068,169 +705,481 @@ def test_get_exploration_stats_multi(self):
+    def test_create_exp_issues_model(self):
+        exp_issues = stats_domain.ExplorationIssues(self.exp.id, 1, [])
 
-        exp_issues = stats_services.get_exp_issues(
-            exploration.id, exploration.version)
-        self.assertEqual(exp_issues.exp_version, 3)
-        self.assertFalse(exp_issues.unresolved_issues[0].is_valid)
-        self.assertEqual(
-            exp_issues.unresolved_issues[0].issue_customization_args[
-                'state_names']['value'], [
-                    'Renamed state', 'End', 'Renamed state'])
+        stats_services.create_exp_issues_model(exp_issues)
 
-        playthrough_instance = stats_models.PlaythroughModel.get(
-            playthrough_id)
-        self.assertEqual(
-            playthrough_instance.issue_customization_args['state_names'][
-                'value'], ['Renamed state', 'End', 'Renamed state'])
-        self.assertEqual(
-            playthrough_instance.actions[0]['action_customization_args'][
-                'state_name']['value'],
-            'Renamed state')
+        exp_issues = (
+            stats_models.ExplorationIssuesModel.get_model(self.exp.id, 1))
+        self.assertEqual(exp_issues.exp_id, self.exp.id)
+        self.assertEqual(exp_issues.exp_version, 1)
+        self.assertEqual(exp_issues.unresolved_issues, [])
+
+    def test_get_exp_issues_creates_new_empty_exp_issues_when_missing(self):
+        exp_issues = (
+            stats_services.get_exp_issues(self.exp.id, self.exp.version))
+        self.assertEqual(exp_issues.exp_id, self.exp.id)
+        self.assertEqual(exp_issues.unresolved_issues, [])
 
     def test_delete_playthroughs_multi(self):
         playthrough_ids = [
-            stats_models.PlaythroughModel.create(
-                'exp_id1', 1, 'EarlyQuit', {}, []),
-            stats_models.PlaythroughModel.create(
-                'exp_id1', 1, 'EarlyQuit', {}, []),
+            self._create_eq_playthrough('A'),
+            self._create_cst_playthrough(['A', 'B', 'A']),
+            self._create_mis_playthrough('A', 3),
         ]
 
         stats_services.delete_playthroughs_multi(playthrough_ids)
 
         instances = stats_models.PlaythroughModel.get_multi(playthrough_ids)
-        self.assertEqual(instances, [None, None])
+        self.assertEqual(instances, [None, None, None])
 
-    def test_get_multiple_exploration_stats_by_version_with_invalid_exp_id(
-            self):
-        exp_stats = stats_services.get_multiple_exploration_stats_by_version(
-            'invalid_exp_id', [1])
+    def test_save_exp_issues_model_transactional(self):
+        eq_playthrough_ids = [
+            self._create_eq_playthrough('End'),
+        ]
+        cst_playthrough_ids = [
+            self._create_cst_playthrough(
+                ['Introduction', 'End', 'Introduction']),
+        ]
+        mis_playthrough_ids = [
+            self._create_mis_playthrough('End', 3),
+        ]
+        stats_services.save_exp_issues_model_transactional(
```

Comment:
This name confused me--typically we only name the thing running in transaction 'transactional' not the thing creating the transaction. Is it much work to update the name of this in the PR? If so, maybe in a follow-up?

Reply:
Err sorry, what's the request here? To remove transactional from the name?

Reply:
Ah, yes. I wasn't very clear earlier. I would expect this to be called ``save_exp_issues_model`` and internally it might have something like ``_save_exp_issues_model_transactional`` that's passed to ``run_in_transaction``. The fact that the public-facing function is run in a transaction is an implementation detail and usually not relevant to the caller (and thus doesn't really belong in the documentation or the name of method).

Reply:
Done.